### ISSUE TITLE
Any tips on how to write custom shell-scripts?

### ISSUE THREAD

**Issue by poppingtonic** [codito/gnome-pomodoro#275]
I can't see any documentation.

**Comment by psukys** [comment]
Can you elaborate on what "custom shell-scripts" are in the scope of this project?

**Comment by kamilprusko** [comment]
Commands work pretty much the same as custom keyboard shortcuts in your system settings, only commands are executed on timer events.

A command need to point to an executable (optionally with params), shell syntax like ```&&```, ```>>``` will be ignored / treated as text. If you want to use bash you need to write your own scripts.

You can pass variables.

To start playing around I suggest adding a command

    echo $(state) ":" $(triggers)

toggle all triggers, and run gnome-pomodoro in a terminal

    $ gnome-pomodoro --quit
    $ gnome-pomodoro

**Comment by kamilprusko** [comment]
I don't think that putting info in a wiki or website will be effective. Ideally, this feature should be self-explanatory - I'm open to suggestions. Or maybe add usual help window/pages in [Mallard](http://www.projectmallard.org/)?

**Comment by NicolasWebDev** [comment]
Is there any update on this one? On how to trigger scripts on alarms?

**Comment by NicolasWebDev** [comment]
Personally, I think that to begin with a really simple wiki page on the subject would help HUGELY as a last resource, to avoid people from opening any more issues on this topic.
Once this is done, maybe it could be linked to in the program itself.

Only the small trick you gave above helped me understand how to use it. I had problems with the shell syntax not working, so I moved all the logic to a during_pomodoro.sh and a during_pause.sh.

**Comment by ginkgomzd** [comment]
I was initially stumped because it is a pretty common paradigm that you can manage preferences from the plugin management. A gear icon next to enable/disable, for instance.

Finally, I noticed that enabling the plugin adds command management to the main preferences.

**Comment by dmhowcroft** [comment]
Thank you @ginkgomzd! I also missed this detail. I wish there was more documentation for this :)

**Comment by josehbez** [comment]
Hi @poppingtonic . I write a simple custom action [gnome-pomodoro-tracking](https://github.com/josehbez/gnome-pomodoro-tracking) that time tracking  with external providers.   I hope it will be of great help.

**Comment by subhasishdas159** [comment]
it's still relevant

**Comment by josehbez** [comment]
I don't think so.
